src/main/scala/firrtl/graph/DiGraph.scala

```diff
@@ -203,36 +203,59 @@ class DiGraph[T] (val edges: Map[T, Set[T]]) extends DiGraphLike[T] {
     val lowlinks = new mutable.HashMap[T, BigInt]
     val sccs = new mutable.ArrayBuffer[Seq[T]]
 
-    def strongConnect(v: T): Unit = {
-      indices(v) = counter
-      lowlinks(v) = counter
-      counter = counter + 1
-      stack.push(v)
-      onstack += v
-      for (w <- getEdges(v)) {
-        if (!indices.contains(w)) {
-          strongConnect(w)
+    /*
+     * Recursive code is transformed to iterative code by representing
+     * call stack info in an explicit structure. Here, the stack data
+     * consists of the current vertex, its currently active edge, and
+     * the position in the function. Because there is only one
+     * recursive call site, remembering whether a child call was
+     * created on the last iteration where the current frame was
+     * active is sufficient to track the position.
+     */
+    class StrongConnectFrame[T](val v: T, val edgeIter: Iterator[T], var childCall: Option[T] = None)
+    val callStack = new mutable.Stack[StrongConnectFrame[T]]
+
+    for (node <- getVertices) {
+      callStack.push(new StrongConnectFrame(node,getEdges(node).iterator))
+      while (!callStack.isEmpty) {
+        val frame = callStack.top
+        val v = frame.v
+        if (frame.childCall.isEmpty) {
+          indices(v) = counter
+          lowlinks(v) = counter
+          counter = counter + 1
+          stack.push(v)
+          onstack += v
+        }
+        for (w <- frame.childCall) {
```

Comment:
Unless I'm very much mistaken, these two branches are mutually exclusive. Perhaps this should be made more explicit with just a match?